♻️ refactor(refactor-user-api): jwtFilter 분리 및 로그인 로그아웃 테스트 코드 추가 (NAILCASE-404)

hidongmin37 opened the pull request:
# Pull Request

[Contribution Guidelines](https://github.com/mobi-projects/nail-case-server/blob/main/WIKI/workflow/guidelines.md)

## PR 제목

♻️ refactor(refactor-user-api): jwtFilter 분리 및 로그인 로그아웃 테스트 코드 추가 (NAILCASE-404)

## PR 설명

-  jwtFilter 분리
- 로그인 로그아웃 테스트 코드 추가

## 관련 이슈

- Resolve: [NAILCASE-456](https://nailcase.atlassian.net/browse/NAILCASE-404)

## 체크리스트

- [x] 코드가 올바르게 동작하는지 확인했습니다.
- [x] 관련 문서가 없습니다.
- [x] 관련 테스트를 추가했습니다.

## 스크린샷 (선택사항)

변경 사항과 관련된 스크린샷을 여기에 추가하세요.

## 추가 정보 (선택사항)

여기에 PR에 대한 추가 정보를 기입하세요.

hidongmin37 committed:
♻️ refactor(refactor-user-api): userPrincipalDetails 생성 (NAILCASE-404)

- memberDetails와 nailAritstDetails통일
- userPrincipalDetails로 생성

resolves: [NAILCASE-404](https://nailcase.atlassian.net/browse/NAILCASE-404)

hidongmin37 committed:
♻️ refactor(refactor-user-api): jwtFilter 분리 (NAILCASE-404)

- jwtFilter 분리

resolves: [NAILCASE-404](https://nailcase.atlassian.net/browse/NAILCASE-404)

hidongmin37 committed:
✅ test(refactor-user-api): user로그인 및 로그아웃 테스트 추가

- 로그인 로그아웃 테스트 추가
- social 로그인 테스트 추가

resolves: [NAILCASE-404](https://nailcase.atlassian.net/browse/NAILCASE-404)

qushe8r commented on src/main/java/com/nailcase/model/dto/UserPrincipalDetails.java:
```diff
@@ -8,52 +8,51 @@
 
 @Getter
-public final class MemberDetails extends User implements UserDetails, UserPrincipal {
+public final class UserPrincipalDetails extends User implements UserDetails, UserPrincipal {
```
User 를 상속하신 이유가 있을까요?
User를 상속하면 User가 이미 구현하고 있기때문에 UserDetails를 구현하지 않아도 괜찮을거에요.

UserPrincipal이 UserDetails를 상속하고 UserPrincipal만 써도 되지 않을까요?
인증 객체로 사용하려고 선언하는 건데 필요한 기능을 추가해서 인터페이스 확장하는 편이 타입 통일하는게 좋지 않을까요?

qushe8r commented on src/main/java/com/nailcase/model/entity/NailArtist.java:
```diff
@@ -34,7 +35,7 @@
 @NoArgsConstructor(access = AccessLevel.PROTECTED)
 @AllArgsConstructor(access = AccessLevel.PROTECTED)
-public class NailArtist extends BaseEntity {
+public class NailArtist extends BaseEntity implements UserPrincipal {
```
NailArtist 와 Member 가 SecurityContext에 인증객체로 사용되나요?

hidongmin37 committed:
✅ test(refactor-user-api): userPrincipleImpl 구현

- userPrinciple,userDetials 에서 중복 코드 및 타입 맞추기

resolves: [NAILCASE-404](https://nailcase.atlassian.net/browse/NAILCASE-404)

hidongmin37 commented:
지금 UserPrincipal쪽을 말씀해주신 것처럼 합치긴 했는데, 인증 인가 필터에서 소셜로그인이 중간에 있어서 하는 부분이 어렵더라구요 억지로 해야되는 부분이 있어서 아마 저쪽도 그부분을 했다가 지운게 이 부분이 클 것 같아요

hidongmin37 commented:
그리고 kakao manager와 kakao member 부분에서도 합치려고 했는데, 이부분에서 결국에 조건문으로 분기가 일어나야하는 부분이 있더라구요 그래서 확장성을 위해서는 분리해두는 것도 좋을 것 같다는 생각에 두긴했습니다

nickkies commented on src/test/java/com/nailcase/controller/AuthControllerTest.java:
```diff
@@ -0,0 +1,89 @@
+		LoginResponseDto actualResponse = authController.socialLogin("member", "kakao", code);
+
+		assertNotNull(actualResponse, "null값이 될 수 없다.");
+		assertEquals(expectedResponse, actualResponse);
```
`@EqualsAndHashCode` 를 `LoginResponseDto`에 넣어줘야 안에 있는 내용까지 비교할 수 있을 것 같아요!

hidongmin37 replied:
넵! 확인해서 수정하겠습니다!

nickkies commented on src/test/java/com/nailcase/service/KakaoManagerLoginServiceTest.java:
```diff
@@ -0,0 +1,74 @@
+		OAuthAttributes attributes = mock(OAuthAttributes.class);
+		when(attributes.getOauth2UserInfo()).thenReturn(oAuth2UserInfo);
+
+		NailArtist nailArtist = NailArtist
+			.builder()
+			.email("test@example.com")
+			.nailArtistId(1L)
+			.role(Role.MANAGER).build();
```
`Fixture` 사용하면 더 좋을 것 같아요!

hidongmin37 replied:
아하 좋습니다!

nickkies reviewed (approved):
고생하셨습니다!

`@AuthenticationPrincipal` 여기서 shopId 들어가는거 관련해서는 일단 홀드 상태인걸까요?

qushe8r reviewed (approved):
수고하셨습니다.
